File: todolist/src/app/accessoryClasses/task/TaskBuilder.ts
/**
 * Created by Павел on 24-Feb-17.
 */
import * as moment from 'moment';
import Moment = moment.Moment;
import {Task} from "./Task";

export class TaskBuilder {
  id: string;
  title: string;
  date: Moment;
  isDone: boolean = false;
  priority:number=4;

  setId(id){
    this.id=id;
    return this
  }

  setTitle(title) {
    this.title=title;
    return this;
  }

  setDate(date){
    this.date=date;
    return this;
  }

  setIdDone(isDone) {
    this.isDone=isDone;
    return this;
  }

  setPriority(priority){
    this.priority=priority;
    return this;
  }

  build() {
    return new Task(this.title,this.date,this.priority,this.isDone,this.id);
  }
}

File: todolist_server/src/main/java/by/zarabon/serverFormats/builders/TaskBuilder.java
package by.zarabon.serverFormats.builders;

import by.zarabon.serverFormats.Task;

public final class TaskBuilder {
    private String id;
    private String date;
    private String title;
    private Integer priority;
    private boolean isDone=false;

    private TaskBuilder() {
    }

    public static TaskBuilder aTask() {
        return new TaskBuilder();
    }

    public TaskBuilder id(String id) {
        this.id = id;
        return this;
    }

    public TaskBuilder date(String date) {
        this.date = date;
        return this;
    }


    public TaskBuilder title(String title) {
        this.title = title;
        return this;
    }

    public TaskBuilder priority(Integer priority) {
        this.priority = priority;
        return this;
    }

    public TaskBuilder isDone(boolean isDone) {
        this.isDone = isDone;
        return this;
    }

    public Task build() {
        Task task = new Task(id,date,title,priority,isDone);
        return task;
    }
}
